### ISSUE TITLE
Scheduler: Observe/Improve event handling throughput 

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes/kubernetes#124566]
/kind feature
/assign
/sig scheduling

---

We don't have any observability around event handling (= requeueing) throughput. Adding observability can be a monitor for a degradation around it. 
Also, based on the metric we add, we can find how fast event handling should be on a certain large scale, like we have 300 pods/s target in scheduling throughput. And then we may or may not need to improve the event handling throughput accordingly.

```[tasklist]
### Tasks
- [x]  1. Add an observability for event handling throughput. https://github.com/kubernetes/kubernetes/pull/125929
- [ ]  2. Benchmark and decide the ideal throughput.
- [ ]  3. Improve the throughput based on (2) if necessary. (possible actions: simplify slower QHint(s), etc)
```

### Reference

- spown from the discussion in https://github.com/kubernetes/kubernetes/issues/124384
- general QHint issue: https://github.com/kubernetes/kubernetes/issues/118893

**Comment by sanposhiho** [comment]
Registered as [Beta requirement: QueueingHint](https://github.com/kubernetes/kubernetes/issues/122597).

**Comment by utam0k** [comment]
I'd like to take part in this issue if possible. Is there any good issue for newer ones?

**Comment by sanposhiho** [comment]
@utam0k 

Let me dig in first. 
Later maybe I'll be able to split it into some tasks that can be done in parallel; then, I'll let you know here.

**Comment by utam0k** [comment]
Sure!

**Comment by sanposhiho** [comment]
Memo: will improve an integration test with a metric:
https://github.com/kubernetes/kubernetes/pull/122234#discussion_r1597456808

cc @AxeZhan

**Comment by sanposhiho** [comment]
/assign @macsko 
/unassign @sanposhiho 

I'm not working on it primarily.
